embrace-android-delivery/src/main/kotlin/io/embrace/android/embracesdk/internal/delivery/storage/PayloadStorageServiceImpl.kt

```diff
@@ -1,13 +1,82 @@
 package io.embrace.android.embracesdk.internal.delivery.storage
 
-import io.embrace.android.embracesdk.internal.comms.delivery.CacheService
+import io.embrace.android.embracesdk.internal.delivery.StoredTelemetryMetadata
+import io.embrace.android.embracesdk.internal.delivery.storedTelemetryComparator
 import io.embrace.android.embracesdk.internal.injection.SerializationAction
+import io.embrace.android.embracesdk.internal.telemetry.errors.InternalErrorService
+import java.io.File
+import java.io.InputStream
+import java.util.zip.GZIPOutputStream
 
 internal class PayloadStorageServiceImpl(
-    private val cacheService: CacheService
+    private val internalErrorService: InternalErrorService,
+    outputDir: Lazy<File>,
+    private val storageLimit: Int = 500 // TODO: check limit
 ) : PayloadStorageService {
 
-    override fun store(filename: String, action: SerializationAction) {
-        cacheService.cachePayload(filename, action)
+    private val payloadDir by outputDir
+
+    // TODO: synchronisation
+
+    override fun store(metadata: StoredTelemetryMetadata, action: SerializationAction) {
+        try {
+            storeImpl(metadata, action)
+        } catch (exc: Throwable) {
+            internalErrorService.handleInternalError(exc)
+        }
+    }
+
+    private fun storeImpl(
```

Comment:
For files that can be overwritten, are you planning to reuse the existing EmbraceCacheService or port them over here too?

Reply:
I was initially planning on moving caching to the `PayloadCachingService` & have the resurrection service send them to the intake service.